### DISCUSSION TITLE
Please help debugging touchpad scrolling

### DISCUSSION THREAD

**Discussion by tothxa** [widelands/widelands#5367] [Q&A]
(sorry for the botched poll... it wasn't meant to be posted yet, and I couldn't figure out how to add more questions)

Please see the discussion of PR #5364

If you have a touchpad, and especially if you have a non-linux system, please try the map scrolling option in the current development version, and report whether horizontal and vertical scrolling are consistent, ie. whether both move the viewport over the map ("traditional" mode) or the map under the viewport ("natural" mode).

Please report with the type of your OS (in case of linux or BSD also the graphical environment – X11/wayland/xwayland), SDL version, and whether map movement is consistent with both system settings of "traditional" and "natural".

(mine is linux / X11 / SDL 2.0.9 and it is consistent with both system settings)

**Comment by attet** [comment]
Windows 11 / SDL 2.0.20#3 / horizontal scrolling is **inconsistent** on both traditional and natural, vertical works as expected

**Comment by Noordfrees** [comment]
My system: Ubuntu MATE 22.04 with X11 and SDL 2.0.20+dfsg-2build1

With "natural scrolling" disabled, vertical touchpad scrolling moves the viewport and horizontal scrolling moves the map. Enabling "natural scrolling" inverts both the horizontal and the vertical scroll direction. So in both cases, horizontal is inconsistent and vertical behaves fine.

**Comment by tothxa** [comment]
The evaluation of SDL version in the flatpak build with runtime 22.08 fails. It thinks it has the inverted x bug, but it shouldn't, because it has version 2.24.0, where `SDL_MINOR == 24` should short out the other checks, keeping `sdl_scroll_x_bug == false`. Yet, the tech info reports:

> Version: 1.1~git25862 (cdb5ccf@master) Release  
> Operating System: Linux  
> Compiled with SDL version: 2.24.0  
> SDL video driver: x11  
> SDL horizontal scroll: assuming inverted  

And indeed, in the Options or Addons dialogs (which don't [segfault](https://github.com/widelands/widelands/issues/5540) :smiley: ) the horizontal scrolling direction is wrong.

**Reply by tothxa** [reply]
Tried to track it down in gdb, and it turns out that the SDL version detection is OK. The problem is that `MousewheelOptionID::kInvertedXDetected` is initialised from the config file (ouch… I just wanted to store the last value as a debug help…), so on upgrading from an old SDL version the compensation remains turned on.
